game-core/src/main/java/games/strategy/engine/data/MoveDescription.java

```diff
@@ -0,0 +1,77 @@
+package games.strategy.engine.data;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.HashMultiset;
+import com.google.common.collect.Maps;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.Getter;
+
+/** Describes an action that moves one or more units along a specific route. */
+public class MoveDescription extends AbstractMoveDescription {
+  private static final long serialVersionUID = 2199608152808948043L;
+  @Getter private final Route route;
+  @Getter private final Map<Unit, Unit> unitsToTransports;
+  @Getter private final Map<Unit, Collection<Unit>> dependentUnits;
```

Comment:
Have you considered using Guava `Multimap`? https://guava.dev/releases/23.0/api/docs/com/google/common/collect/Multimap.html

It is essentially the same datastructure as `Map<X Collection<Y>>`.

Reply:
I'd rather not make this change bigger than it is. This type is already in use throughout the code.

Reply:
Sounds okay, didn't quite realize how much of this class was just a move vs update.